editoast/src/views/path/pathfinding.rs

```diff
@@ -73,6 +73,45 @@ struct PathfindingInput {
     rolling_stock_length: OrderedFloat<f64>,
 }
 
+impl <'de> Deserialize<'de> for PathfindingInput {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de> {
+        #[derive(Deserialize)]
+        #[serde(deny_unknown_fields)]
+        struct Internal {
+            rolling_stock_loading_gauge: LoadingGaugeType,
+            rolling_stock_is_thermal: bool,
+            rolling_stock_supported_electrifications: Vec<String>,
+            rolling_stock_supported_signaling_systems: Vec<String>,
+            path_items: Vec<PathItemLocation>,
+            rolling_stock_maximum_speed: OrderedFloat<f64>,
+            rolling_stock_length: OrderedFloat<f64>,
+        }
+        let Internal {
+            rolling_stock_loading_gauge,
+            rolling_stock_is_thermal,
+            rolling_stock_supported_electrifications,
+            rolling_stock_supported_signaling_systems,
+            path_items,
+            rolling_stock_maximum_speed,
+            rolling_stock_length,
+        } = Internal::deserialize(deserializer)?;
+        if path_items.first() == path_items.last() {
```

Comment:
This is not enough to figure out if the path length will be zero. For instance, a UIC+ch could refer to the same OP as a trigram+ch.

Reply:
I am missing some knowledge about what a trigram and an UIC represent and how to correlate them. Would you mind to elaborate or hint at some documentation on them ?

Reply:
Sure. There are multiple ways to refer to an operational point in a path: https://github.com/OpenRailAssociation/osrd/blob/9dedb376fde0341f4a6d6f94090dc1044c90d523/editoast/editoast_schemas/src/train_schedule/path_item.rs#L64

One way is via a _UIC code_. The UIC code is a worldwide-unique number which identifies a location. For instance, "686006" is "Paris-Gare-de-Lyon". In France, the UIC is not enough to precisely identify an operational point: there are multiple operational points in "Paris-Gare-de-Lyon". So a _secondary code_ is added to pin down a specific one (often also called "ch" for "code chantier"). For instance, "BV" often refers to the place where the platforms are in a station ("bâtiment voyageur"). Side note, the UIC is made up of two parts, a country prefix ("68" for France) and the rest is called _CI_ ("code immuable", local French code).

Another way is via a _trigram_. It's essentially another way to write a CI, perhaps easier to remember for humans. For instance, "PLY" is the trigram for "Paris-Gare-de-Lyon".

Yet another way is via an _operational point ID_. It's just an opaque string identifier (e.g. "1e083f8b-c7db-48f4-bb12-ac3bb21d8963").

So, both of these refer to the same operational point:

```rust
OperationalPointDescription {
    trigram: "PLY",
    secondary_code: "BV",
}
OperationalPointUic {
    uic: 686006,
    secondary_code: "BV",
}
```

In other words, if these are given as input to the pathfinding algorithm, it'll freak out the same way it would if given two exact same path item locations.

Reply:
Thank you for the explanation ! I added a new error variant `ZeroLengthPath` that transforms `core` success responses with zero-length paths into errors instead of checking whether the origin and destination are the same location before executing the pathfinding.